Review CIS RHEL8 v3.0.0 Section 4 and Section 5

marcusburghardt opened the pull request:
#### Description:

Review the cis_rhel8.yml control file and update the following sections  in alignment to CIS RHEL 8 v3.0.0:
- Section 4 - Access, Authentication and Authorization
- Section 5 - Logging and Auditing

Also update references in related rules.

#### Rationale:

Keep RHEL 8 profiles updated with CIS RHEL 8 last version.

#### Review Hints:

NOTE: Many requirements were swapped between section 4 and 5. This is the reason this PR includes the review of both sections.

The easiest way is to open the CIS RHEL8 v3.0.0 policy and compare the changes in commits with the respective requirements.

marcusburghardt commented:
The conflicts here are basically the same I had to resolve in https://github.com/ComplianceAsCode/content/pull/11455.
I will wait https://github.com/ComplianceAsCode/content/pull/11455 to be merged first then I rebase this PR.

marcusburghardt commented:
I rebased this PR on top of the latest upstream master branch.

marcusburghardt commented:
Rebased again.

vojtapolasek reviewed (changes requested):
Hello @marcusburghardt  and thank you for this big update. I have only minor comments, it looks mostly good.

marcusburghardt commented on controls/cis_rhel8.yml:
```diff
@@ -1531,298 +1531,821 @@ controls:
     rules:
-      - audit_rules_suid_auid_privilege_function
+      - file_at_deny_not_exist
+      - file_groupowner_at_allow
```
In combination with the rule to ensure there is no at.deny file I consider enough to satisfy this requirement. But I think it is worth for a note. I will include a note here.

marcusburghardt replied:
Done.

marcusburghardt commented on controls/cis_rhel8.yml:
```diff
@@ -1531,298 +1531,821 @@ controls:
-      - audit_rules_usergroup_modification_opasswd
-      - audit_rules_usergroup_modification_passwd
-      - audit_rules_usergroup_modification_shadow
+      - sshd_use_approved_ciphers
```
Great

marcusburghardt replied:
Done

marcusburghardt commented on controls/cis_rhel8.yml:
```diff
@@ -1531,298 +1531,821 @@ controls:
+      - l1_workstation
+    status: pending
+    notes: |-
+      Rule sshd_use_strong_kex needs to be enabled for rhel8.
```
For sure. I was not sure about this some days ago. : ) I am updating now.

marcusburghardt replied:
Done

marcusburghardt commented on controls/cis_rhel8.yml:
```diff
@@ -1531,298 +1531,821 @@ controls:
+      - l1_workstation
+    status: pending
+    notes: |-
+      Rule sshd_use_strong_macs needs to be enabled for rhel8.
```
done.

marcusburghardt commented on controls/cis_rhel8.yml:
```diff
@@ -1531,298 +1531,821 @@ controls:
+      - l1_workstation
+    status: automated
+    rules:
+      - accounts_passwords_pam_faillock_unlock_time
```
Its a fair point. I will include a note.

marcusburghardt replied:
Done

marcusburghardt commented on controls/cis_rhel8.yml:
```diff
@@ -1531,298 +1531,821 @@ controls:
+      - accounts_umask_etc_bashrc
+      - accounts_umask_etc_login_defs
+      - accounts_umask_etc_profile
+      - accounts_umask_interactive_users
```
Aligned.

marcusburghardt commented on controls/cis_rhel8.yml:
```diff
@@ -1531,298 +1531,821 @@ controls:
-      - l2_workstation
+      - l1_server
+      - l1_workstation
     status: automated
```
Done

marcusburghardt commented on controls/cis_rhel8.yml:
```diff
@@ -1981,564 +2495,442 @@ controls:
-      - l1_workstation
+      - l2_server
+      - l2_workstation
     status: automated
```
Done.

marcusburghardt commented:
> Hello @marcusburghardt and thank you for this big update. I have only minor comments, it looks mostly good.

Thanks for the review @vojtapolasek . I addressed all the comments in the last commits.

vojtapolasek commented:
Looks great now @marcusburghardt , just waiting for CI to finish.

vojtapolasek commented:
Waiving tests on Rawhide, Rawhide is currently broken. The PR does not touch anything related to Kubernetes so waiving as well.
